[FBZ-10448] - Migrate New Relic Infra Recipe to v7

kdotbautista opened the pull request:
**Description of your patch**
Adds support for New Relic Infra by migrating recipe from v6 to v7

**Recommended Release Notes**
Added support for New Relic Infra

**Estimated risk**
Medium

**Components involved**
`cookbooks/ey-newrelic_infra`
`custom-cookbooks/newrelic_infra`


**Description of testing done**

* Create a new environment with v7 as stack
* Upload recipe for `cookbooks/ey-newrelic_infra` and `custom-cookbooks/newrelic_infra`
* Verify through the chef logs that ey-newrelic_infra recipe has been executed
* Verify if New Relic Package has been successfully installed

**QA Instructions**

* Create a new environment with v7 as stack
* Upload recipe for `cookbooks/ey-newrelic_infra` and `custom-cookbooks/newrelic_infra`
* Verify through the chef logs that ey-newrelic_infra recipe has been executed
* Verify if New Relic Package has been successfully installed

mushyy commented on cookbooks/ey-newrelic_infra/recipes/default.rb:
```diff
@@ -0,0 +1,38 @@
+  })
+end
+
+execute "curl -s https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg | sudo apt-key add -" do
```
Keys should be added under apt_repository process

kvbautista replied:
Moved key under apt_repository process

```
apt_repository "newrelic-infra" do
  uri "https://download.newrelic.com/infrastructure_agent/linux/apt"
  distribution "focal"
  distribution "#{`lsb_release -cs`.strip}"
  components ["main"]
  key "https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg"
end.run_action(:add)
```

mushyy commented on cookbooks/ey-newrelic_infra/files/default/newrelic-infra.list:
```diff
@@ -0,0 +1 @@
+deb [arch=amd64] https://download.newrelic.com/infrastructure_agent/linux/apt focal main
```
This file should be generated when adding also, the type should support arm64 as well as amd64

kvbautista replied:
Removed files folder as `newrelic-infra.list` should be generated upon execution of `apt_repository` process

kdotbautista commented:
Moved key under apt_repository process for `cookbooks/ey-newrelic_infra/recipes/default.rb`

```
apt_repository "newrelic-infra" do
  uri "https://download.newrelic.com/infrastructure_agent/linux/apt"
  distribution "focal"
  distribution "#{`lsb_release -cs`.strip}"
  components ["main"]
  key "https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg"
end.run_action(:add)
```

And removed files folder as newrelic-infra.list should be generated upon execution of `apt_repository` process.

mushyy reviewed (commented):
Looks good to me passes linting process

mushyy commented:
This will be merged for next release
